packages/plugins/foreman-discovery-image-service/foreman-discovery-image-service.spec

```diff
@@ -40,7 +47,20 @@ Metapackage with dependencies for FDI text-user interface
 
 %files tui
 
+%files dracut
+
+%pre dracut
+# Because livecd-creator executes %post scriptlets after it copies initramdisk out of the
+# image, it is not possible to rebuild kernel ramdisk with extra drivers for VMWare and
+# MS Hyper-V. Therefore the configuration must be dropped before the kernel package
+# executes its %post scriptlet. Doing it in %pre ensures the configuration is created
+# beforehand.
+echo 'add_drivers="mptbase mptscsih mptspi hv_storvsc hid_hyperv hv_netvsc hv_vmbus"' > %{_sysconfdir}/dracut.conf.d/99-discovery.conf
```

Comment:
Why is this a post script vs just a file shipped in the package?

Reply:
So correct me if I am wrong, but the goal is to have that configuration in-place before kernel's %post scriptlet is executed. Since %post scriptlets are executed one by one as packages are installed I cannot guarantee the order (because kernel can/will likely install before this package).

This actually makes me think if I should actually change this to `%pretrans` that is guaranteed to be executed before "transaction", whatever that actually is: https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/

Reply:
Wait, what, the result is dependant on the order dnf decides to install the packages in? What the actual fuck?!

No, I am not going to think about this at this time.

If you're going to edit the file in any scriptlet, please add a `%ghost` entry to the files section, so rpm at least knows who owns the file.

Reply:
I have found a solution, %triggerprein is exactly what I need. I am gonna try to test that.

BTW do I miss buildroot on this line? I guess this is correct:

```
echo 'xxx' > %{buildroot}%{_sysconfdir}/dracut.conf.d/99-discovery.conf
```

Reply:
Damn this must NOT have a buildroot macro of course! :) Jeez and its Monday.

https://pbs.twimg.com/media/FN9eS68X0AIVaqv.jpg